src/BaGetter/Program.cs

```diff
@@ -7,14 +8,26 @@
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Hosting.WindowsServices;
 
 namespace BaGetter;
 
 public class Program
 {
     public static async Task Main(string[] args)
     {
-        var host = CreateHostBuilder(args).Build();
+        if (WindowsServiceHelpers.IsWindowsService())
+        {
+            // Make sure appsettings.json can be found when running as windows service
+            Directory.SetCurrentDirectory(AppDomain.CurrentDomain.BaseDirectory);
+        }
+
+        var builder = CreateHostBuilder(args);
+
+        // Support running as a windows service
+        builder.UseWindowsService();
+
+        var host = builder.Build();
```

Comment:
I think something like this would make more sense:

```suggestion
        var builder = CreateHostBuilder(args);

        // Support running as a windows service
        if (WindowsServiceHelpers.IsWindowsService())
        {
            // Make sure appsettings.json can be found when running as windows service
            Directory.SetCurrentDirectory(AppDomain.CurrentDomain.BaseDirectory);

            builder.UseWindowsService();
        }

        var host = builder.Build();
```

Reply:
Hi @seriouz, 

Thank you for commenting and proposing improvements.

Wrapping builder.UseWindowsServices() into an "if( IsWindowsService )" is not necessary.
The extension method already checks this internally.

Also if you want to chain it with UseSystemd() it would not make sense, e.g.

`builder.UseWindowsService().UseSystemd()`

Regards the order when to apply the SetCurrentDirectory(...) code.
I think it needs to happen before the CreateHostBuilder() call (as it was in my code snippet) because otherwise appsettings.json is not found correctly during CreateHostBuilder(...) and e.g. the kestrel URL config in appsettings.json is not applied correctly